### PR TITLE
`gw-disable-submit-button-until-required-fields-are-filled-out.php`: Fixed multiple file upload compatibility with Disable Submit snippet.

### PR DESCRIPTION
**Pull request by saifsultanc**
## Context

⛑️ Ticket(s): https://secure.helpscout.net/conversation/2570453216/64916/

## Summary

The [Disable Submit snippet](https://github.com/gravitywiz/snippet-library/blob/master/gravity-forms/gw-disable-submit-button-until-required-fields-are-filled-out.php) doesn't work with a required Upload field, with mutli-file enabled. This adds compatibility for that.

### CODE REVIEW

File: gravity-forms/gw-disable-submit-button-until-required-fields-are-filled-out.php [hunk 1]
```diff
@@ -64,6 +64,19 @@ public function script() {
...
+						if ( fileUploadInput ) {
+							const observer = new MutationObserver( function( mutationsList, observer ) {
+								mutationsList.forEach( mutation => {
+									if ( mutation.type === 'attributes' && mutation.attributeName === 'value' ) {
+										self.runCheck();
+									}
+								});
+							});
+							observer.observe( fileUploadInput, { attributes: true });
+						}
```
[the remark this answers is not in the document]
Comment (saifsultanc) [reply gravity-forms/gw-disable-submit-button-until-required-fields-are-filled-out.php:78]: `StateChanged` is fine, but the problem is when we remove files I am not able to tap any of those events on `gform_multifile_upload_`

I tested, following code update:

```php
						args.inputHtmlIds.forEach(function (inputHtmlId){
							let searchTerm = "#gform_preview";
							let index	   = inputHtmlId.indexOf(searchTerm);
							if ( index !== -1 ) {
								let fieldId = inputHtmlId.substring(index + searchTerm.length).trim();
								window.gfMultiFileUploader.uploaders['gform_multifile_upload' + fieldId].bind('StateChanged', function() {
									if ( window.gfMultiFileUploader.uploaders['gform_multifile_upload' + fieldId].state == 1 ) {
										self.runCheck();
									}
								});
							}
						});
```

I tried `FilesRemoved` but not able to work with it in this context. `MO` works fine overall, that's why I had opted for it.

[the remark this answers is not in the document]
Comment (claygriffiths) [reply gravity-forms/gw-disable-submit-button-until-required-fields-are-filled-out.php:78]: Ah, right. Gravity Forms doesn't run through Plupload when removing files. Let me do some more digging.

[the remark this answers is not in the document]
Comment (claygriffiths) [reply gravity-forms/gw-disable-submit-button-until-required-fields-are-filled-out.php:78]: One option would be to add a simple `click` listener on `.gform_delete`.

That said, do we even need to run checks after deleting? All we're really concerned about are in-flight uploads, correct?

[the remark this answers is not in the document]
Comment (saifsultanc) [reply gravity-forms/gw-disable-submit-button-until-required-fields-are-filled-out.php]: Updated (with quick loom):
https://www.loom.com/share/150b13f9e48d452d83ab7c89e3cfb4df

**Review by claygriffiths — APPROVED** [review]
LGTM!

**Pull request merged**